### PR TITLE
fix(es/async-generator): Fix `yield*` in async generators

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue (if exists):**

 - Closes https://github.com/swc-project/swc/issues/4405

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_transforms_compat

File: crates/swc/tests/exec/issues-4xxx/4406/1/exec.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+async function* execute(
+) {
+    yield* asyncGenerator();
+}
+
+(async () => {
+    for await (const p of execute()) {
+        console.log(await p);
+    }
+})();
```
Comment (zEh-) [inline crates/swc/tests/exec/issues-4xxx/4406/1/exec.js:18]: New to this repo, but isn't it #4405 (not #4406)?

Comment (kdy1) [reply crates/swc/tests/exec/issues-4xxx/4406/1/exec.js:18]: You are right, thank you!

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_transforms_base
 - swc_ecma_transforms_compat

**Pull request merged**